testing/travis-fork-check.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import logging
+import os
+import subprocess
+import sys
+
+def check_travis_repo_slug():
+    if 'encrypted_6a9d32f3e0bd_key' not in os.environ:
+        logging.warning("Decryption disabled. Setting MANTL_CI_FORK_CHECK to 1")
+        print("1")
+    else:
+        logging.info("Decrypting OS ssh key")
```

Comment:
Take a look at this error’d build from Travis: https://travis-ci.org/CiscoCloud/mantl/jobs/127433850

In this version, the `openssl` command was run as its own step in `.travis.yml`, but it would error out on PRs from forks. So my solution was to wrap that around a check to see if the encrypted vars were actually available. If they are, then run the decryption, otherwise, skip it.
> On May 11, 2016, at 12:57, Brian Hicks <notifications@github.com> wrote:
> 
> In testing/travis-fork-check.py <https://github.com/CiscoCloud/mantl/pull/1415#discussion_r62895274>:
> 
> > @@ -0,0 +1,30 @@
> > +#!/usr/bin/env python
> > +from __future__ import print_function
> > +import logging
> > +import os
> > +import subprocess
> > +import sys
> > +
> > +def check_travis_repo_slug():
> > +    if 'encrypted_6a9d32f3e0bd_key' not in os.environ:
> > +        logging.warning("Decryption disabled. Setting MANTL_CI_FORK_CHECK to 1")
> > +        print("1")
> > +    else:
> > +        logging.info("Decrypting OS ssh key")
> So… why is decrypting a SSH key in the script to detect if the build is a fork? They ought to be separate.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/CiscoCloud/mantl/pull/1415/files/dce2ccb628c69104764c0cad923e016abacadee1#r62895274>